src/config.ts

```diff
@@ -223,6 +254,8 @@ export async function readConfigs(configPaths: string[]): Promise<UnifiedConfig>
     }
   }
 
+  const directoryPaths = configPaths.map(parseDirectoryPath);
```

Comment:
I had to fix a bunch of downstream logic here since it was built under the assumption that you were always passing a file path where you had to parse out the file name + extension.